ClassRegistration/ClassRegistration.DataAccess/Repository/Repository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using ClassRegistration.DataAccess.Entity;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ClassRegistration.DataAccess.Repository {
+
+    public class Repository {
+
+        protected readonly Course_registration_dbContext _context;

```

Comment:
Is it the same as making the field private?

Reply:
"protected" is shared only to sub-classes. We typically use protected data for the generic Repository base class, because we only make instances of it's child classes. Therefore, it's necessary for the child classes to know about this data, but not the rest of the code